lib/active_merchant/billing/gateways/xpay.rb

```diff
@@ -76,6 +82,23 @@ def scrub(transcript)
 
       private
 
+      def add_capture_type(options, action)
+        post = {}
+        case action
+        when :purchase
+          post[:captureType] = 'IMPLICIT'
+        when :authorize
+          post[:captureType] = 'EXPLICIT'
+        end
+        post
+      end
```

Comment:
The function's name isn't inherently wrong, but it implies adding the capture type to someone, as is the case with `add_refund_capture_params`. 
However, the function returns a new hash with or without values. You might want to consider a name that better reflects the function's behavior.

Reply:
What do you think if you use constants, 

```ruby
   when PaymentActions::PURCHASE
    post[:captureType] = 'IMPLICIT'
  when PaymentActions::AUTHORIZE
    post[:captureType] = 'EXPLICIT'
  end
```
consider it also for the :operation_id setters.

Reply:
Thanks for the suggestion! Changed!